internal/credentials/model.go

```diff
@@ -59,7 +59,7 @@ func (m *Model) Prefix() string {
 }
 
 // NewCredential returns a new credential model
-func NewCredential(username, password string) ICredentials {
+func NewCredential(username, password string) *Model {
```

Comment:
IModel ?

Reply:
Serde was failing with the interface previously hence changed it to the struct{}.